### PR TITLE
Image documentation

### PR DESCRIPTION
**Pull request by soltysh**
This PR contains 2 commits with new python image and unifying changes to other images in a separate commit.

//fyi @jhadvig @mnagy @mfojtik @bparees

### CODE REVIEW

File: openshift_sti_images/nodejs.adoc [hunk 1]
```diff
@@ -122,8 +122,8 @@ If you omit the `VERSION` parameter, the build/test action is performed on all s
 |This script is responsible for using the application web server to run the application.
 
 |`/nodejs-version/.sti/bin/`
-|[filename]#save-artifacts#
-|This script archives all dependent modules in this image in order to perform an incremental build. An incremental build reuses the build artifacts from a previously-built image to create a new image.
+|[filename]#usage#
+|This script is printing the usage of this image.
```
Comment (bparees) [inline openshift_sti_images/nodejs.adoc]: This script prints the usage of this image.

File: openshift_sti_images/python.adoc [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+:toc: macro
+:toc-title:
+
+toc::[]
+
+== Overview
+The OpenShift https://github.com/openshift/sti-python/tree/master/3.3[Python] image repository contains the sources and Dockerfiles for building various versions of Python platform images. The resulting images can be run either by Docker or using https://github.com/openshift/source-to-image[STI].
+
+== Versions
+Currently, OpenShift only supports version https://github.com/openshift/sti-python/tree/master/3.3[3.3] of Python.
```
Comment (bparees) [inline openshift_sti_images/python.adoc]: it's not an openshift support statement...we support basically anything.  would not want someone to read this and think they can't run a python 2.7 image on openshift.

Comment (bparees) [inline openshift_sti_images/python.adoc]: we should strike this wording from our other docs too...  "openshift provides a 3.3 version" would be better.

File: openshift_sti_images/python.adoc [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+
+== Base Images
+
+* RHEL-7
+* CentOS-7
+
+[NOTE]
+====
+In order to perform build or test action on a RHEL-7 based Python image, you need to run the build on properly subscribed RHEL-7 machine.
```
Comment (bparees) [inline openshift_sti_images/python.adoc]: In order to run the build or test action on.....

File: openshift_sti_images/python.adoc [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+----
+
+=== CentOS-7 Base Image
+
+----
+$ git clone https://github.com/openshift/sti-python.git
+$ cd sti-python
+$ make build VERSION=3.3
+----
+
```
Comment (bparees) [inline openshift_sti_images/python.adoc]: "or just pull it from docker hub......"  (again all our docs should say this for the centos images)

Comment (soltysh) [inline openshift_sti_images/python.adoc]: Added.

**Comment by soltysh** [comment]
@bparees addressed your comments in all other images as well. I'm also doing those updates in our README's.

**Comment by jhadvig** [comment]
thanks for updating other docs :+1: 
LGTM

**Comment by bparees** [comment]
lgtm, thanks for doing all the other images!

**Comment by soltysh** [comment]
Yeah I'm bit of a pedantic in those things and I like to keep those in a
similar manner :wink:

**Comment by tpoitras** [comment]
I'm a little confused why a lot of my work has been reversed and undone?

I worked on the supplied text for these images, in order to clean up the supplied text, and bring them into line with proper tech writing style, and proper use of the openshift doc guidelines. 

This work has been removed and reversed back to the state those files were in when they were written by the dev? 

@adellape @CowboysFan @nhr -- do not merge this PR yet.

**Comment by CowboysFan** [comment]
@tpoitras @soltysh @bparees I'm a bit confused myself. If this content applies to OpenShift Enterprise then there are certain CCS style and writing guidelines we have to follow for our Enterprise customers. However, if it's only for Origin then we're happy to leave it as-is. But if that's not the case, then I'm not sure why the changes made by @tpoitras were reverted back. This section and content still must go through a lot more changes in terms of structure and style before it's deemed ready to publish for Enterprise, and @tpoitras had started that process with his previous changes. So I'm not sure what's happened here but it would be good to get this clarified. Thanks.

**Comment by bparees** [comment]
@CowboysFan i'm not familiar w/ our processes for enterprise vs origin docs, how are we distinguishing?  as far as i know all the stuff in openshift-docs is ending up here:  http://docs.openshift.org/latest/welcome/index.html (which would be origin).  I would think that OSE docs would be being kept in a private repo.

I can't speak to the specific changes that may have been reverted in this, we'll have to wait for @soltysh  to weigh in,  but the goal here was to reconcile the way we're documenting the same basic information about different images, so we present the information in the same way in all places.

**Comment by CowboysFan** [comment]
@bparees Thanks for replying. We're using one repo for all OpenShift docs, and we'll distinguish between the different products with separate branches. However, we're trying to single-source as much as possible and reuse as much content as possible. That's why we're also fixing all the content on the 'master' or working branch because as far as I've been told OSE v3.0 will be the same as Origin with the first release. This way Origin can also use the content that we fix up as our contribution to the upstream docs; this is all explained in the README of the docs repo. If there is content that is specific to Origin or OSE within a topic, we'll conditionalize that content so that it only shows up in that particular product. So we won't take a snapshot of the docs until that last minute for OSE. I hope that makes sense, otherwise we can discuss further, or @nhr can also explain since he's in the same timezone as you. :)

**Comment by soltysh** [comment]
@tpoitras sorry if I removed your work, it wasn't intentional. Can you point me which parts of this PR did this? The only I can think of are the comment between commands. The whole purpose of this PR is to unify all our images, because when I was writing the docs for the python image I've noticed big differences and some outdated information as well. I can remove parts of this PR (please show me which) if you can promise me you'll address those differences in your work.

**Comment by CowboysFan** [comment]
@soltysh Thanks for this info and clarification.

**Comment by tpoitras** [comment]
I made a bunch of comments. That's as deep as I'm going to go right now, as I'm actually on PTO. I just didn't want this to merge while I was away. I'll take another look when I'm back to work on April 14th.

**Comment by soltysh** [comment]
@tpoitras take your time,  I'll be on PTO until Apr 17. I'll address all
your comments after that date

**Comment by mnagy** [comment]
FYI, this PR will also resolve bug https://bugzilla.redhat.com/show_bug.cgi?id=1210138
Just so we remember to close it after this is merged.

**Comment by tpoitras** [comment]
@soltysh -- are you going to revert the changes required (indicated by my comments above), or do you want to just create a new PR that only has the python info (and close this PR) so I can get it ready for beta3?

**Comment by soltysh** [comment]
@tpoitras I'm planning to do that today.

**Comment by soltysh** [comment]
@tpoitras as promised updated. I've mainly removed all the destructive changes of mine, sorry once again for doing that. I've also (based on other images) extended other images descriptions when invoking commands. Appreciate review.

**Comment by soltysh** [comment]
I've removed the [do not merge] tag, since @tpoitras is OK with merging this as is. 
@tpoitras thx for your help and sorry for the trouble once again :smile:

**Comment by adellape** [comment]
@soltysh :cool:, merging.

**Pull request merged**